libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipRouter.kt

```diff
@@ -398,7 +415,7 @@ open class GossipRouter(
                     }
                     .flatten()
             }
-        val list = peers.map { submitPublishMessage(it, msg) }
+        val list = peers.filterNot { peerDoesNotWantMessage(it, msg.messageId) }.map { submitPublishMessage(it, msg) }
```

Comment:
I would also add `sendIDontWant()` here. This would be useful when original message is recovered locally and is reseeded to the topic.